eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/JavadocCodeSnippetCheck.java

```diff
@@ -0,0 +1,190 @@
+                break;
+            case TokenTypes.METHOD_DEF:
+                methodDefStack.push(token);
+                break;
+            case TokenTypes.BLOCK_COMMENT_BEGIN:
+                checkNamingPattern(token);
+                break;
+            default:
+                // Checkstyle complains if there's no default block in switch
+                break;
+        }
+    }
+
+    /**
+     * Check if the given block comment is on method. If not, skip the check.
+     * Otherwise, check if the {@literal {@codesnippet}} has matching the naming pattern
+     *
+     * @param blockCommentToken BLOCK_COMMENT_BEGIN token
+     */
+    private void checkNamingPattern(DetailAST blockCommentToken) {
+        if (!BlockCommentPosition.isOnMethod(blockCommentToken)) {
+            return;
+        }
+
+        // Turn the DetailAST into a Javadoc DetailNode.
+        DetailNode javadocNode = null;
+        try {
+            javadocNode = DetailNodeTreeStringPrinter.parseJavadocAsDetailNode(blockCommentToken);
+        } catch (IllegalArgumentException ex) {
+            // Exceptions are thrown if the JavaDoc has invalid formatting.
+        }
+
+        if (javadocNode == null) {
+            return;
+        }
+
+        // Iterate through all the top level nodes in the Javadoc, looking for the @throws statements.
+        for (DetailNode node : javadocNode.getChildren()) {
+            if (node.getType() != JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                continue;
+            }
+            // Skip if not codesnippet annotation
+            DetailNode customNameNode = JavadocUtil.findFirstToken(node, JavadocTokenTypes.CUSTOM_NAME);
+            if (customNameNode == null || !CODE_SNIPPET_ANNOTATION.equals(customNameNode.getText())) {
+                return;
+            }
+            // Missing Description
+            DetailNode descriptionNode = JavadocUtil.findFirstToken(node, JavadocTokenTypes.DESCRIPTION);
+            if (descriptionNode == null) {
+                log(node.getLineNumber(), MISSING_CODESNIPPET_TAG_MESSAGE);
+                return;
+            }
+
+            String description = JavadocUtil.findFirstToken(descriptionNode, JavadocTokenTypes.TEXT).getText();
+
+            // Find method name
+            DetailAST methodDefToken = methodDefStack.peek();
+            final String methodName = methodDefToken.findFirstToken(TokenTypes.IDENT).getText();
+            final String className = classNameStack.isEmpty() ? "" : classNameStack.peek();
+            final String parameters = constructParametersString(methodDefToken);
+            String fullPath = packageName + "." + className + "." + methodName;
+
+            if (parameters != null) {
+                fullPath = fullPath + "#" + parameters;
+            }
+
+            // Check for CodeSnippet naming pattern matching
+            if (!description.equalsIgnoreCase(fullPath)) {
```

Comment:
We should be a bit more lenient here. For example, imagine the situation where a JavaDoc block has two code samples - there would be no way to differentiate them! You should probably do a 'startsWith' check instead, so we can support -1, -2, -3, etc

Reply:
can you give an example? I am not very sure about the case.

Reply:
Imagine the following:

```java
/**
 * Sample code one:
 * {@codesnippet com.azure.example.method}
 *
 * Here's another example:
 * {@codesnippet com.azure.example.method-2}
 */
public void method() { 
...
} 
```